### PR TITLE
[ASCII-1329] Remove some unknown config keys warning logs

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Set some missing configs as `known` to avoid warnings.

### Motivation

The config logs warnings when accessing keys that weren't declared with `SetKnown`, `SetDefault`, `BindEnv` or some other similar methods.
This PR sets some keys as known to remove warnings.

### Additional Notes

The keys that had warnings before which are fixed by this PR are:
- `secret_audit_file_max_size`
- `system_probe_config.adjusted`
- `win_skip_com_init`
- `system_probe_config.process_service_inference.use_improved_algorithm`
- `security_agent.security_agent.internal_profiling.profile_dd_url`
- `runtime_security_config.activity_dump.remote_storage.endpoints.dev_mode_no_ssl`
- `compliance_config.endpoints.dev_mode_no_ssl`
- `container_image.dev_mode_no_ssl`
- `container_lifecycle.dev_mode_no_ssl`
- `database_monitoring.activity.dev_mode_no_ssl`
- `database_monitoring.metrics.dev_mode_no_ssl`
- `database_monitoring.samples.dev_mode_no_ssl`
- `logs_config.dev_mode_no_ssl`
- `network_devices.metadata.dev_mode_no_ssl`
- `network_devices.netflow.forwarder.dev_mode_no_ssl`
- `network_devices.snmp_traps.forwarder.dev_mode_no_ssl`
- `network_path.forwarder.dev_mode_no_ssl`
- `runtime_security_config.endpoints.dev_mode_no_ssl`
- `sbom.dev_mode_no_ssl`

### CODE REVIEW

File: cmd/security-agent/subcommands/start/command.go [hunk 1]
```diff
@@ -385,7 +385,7 @@ func setupInternalProfiling(config config.Component) error {
 		v, _ := version.Agent()
 
 		cfgSite := config.GetString(secAgentKey("internal_profiling.site"))
-		cfgURL := config.GetString(secAgentKey("security_agent.internal_profiling.profile_dd_url"))
+		cfgURL := config.GetString(secAgentKey("internal_profiling.profile_dd_url"))
```
Comment (pgimalac) [inline cmd/security-agent/subcommands/start/command.go:388]: Let me know if the `security_agent` stutter due to `secAgentKey` was on purpose

File: cmd/system-probe/config/adjust.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ var adjustMtx sync.Mutex
 func Adjust(cfg config.Config) {
 	adjustMtx.Lock()
 	defer adjustMtx.Unlock()
-	if cfg.GetBool(spNS("adjusted")) {
+	if cfg.IsKnown(spNS("adjusted")) {
```
Comment (pgimalac) [inline cmd/system-probe/config/adjust.go:24]: Let me known if this change works for you, I could just `SetKnown("adjusted")` too 🤷‍♂️

File: pkg/config/setup/config.go [hunk 3]
```diff
@@ -2041,6 +2043,7 @@ func bindEnvAndSetLogsConfigKeys(config pkgconfigmodel.Config, prefix string) {
 	config.BindEnvAndSetDefault(prefix+"sender_recovery_interval", DefaultForwarderRecoveryInterval)
 	config.BindEnvAndSetDefault(prefix+"sender_recovery_reset", false)
 	config.BindEnvAndSetDefault(prefix+"use_v2_api", true)
+	config.SetKnown(prefix + "dev_mode_no_ssl")
```
Comment (pgimalac) [inline pkg/config/setup/config.go:2046]: Let me know if I should set a default and/or bind environment variable

File: pkg/config/setup/system_probe.go [hunk 4]
```diff
@@ -202,6 +202,8 @@ func InitSystemProbeConfig(cfg pkgconfigmodel.Config) {
 
 	cfg.BindEnvAndSetDefault(join(spNS, "language_detection.enabled"), false)
 
+	cfg.SetKnown(join(spNS, "process_service_inference", "use_improved_algorithm"))
```
Comment (pgimalac) [inline pkg/config/setup/system_probe.go:205]: Let me know if I should set a default and/or bind environment variable

File: pkg/config/setup/config.go [hunk 5]
```diff
@@ -365,6 +366,7 @@ func InitConfig(config pkgconfigmodel.Config) {
 	config.BindEnvAndSetDefault("secret_backend_skip_checks", false)
 	config.BindEnvAndSetDefault("secret_backend_remove_trailing_line_break", false)
 	config.BindEnvAndSetDefault("secret_refresh_interval", 0)
+	config.SetKnown("secret_audit_file_max_size")
```
Comment (pgimalac) [inline pkg/config/setup/config.go:369]: Let me know if I should set a default and/or bind environment variable

File: pkg/config/setup/config.go [hunk 6]
```diff
@@ -258,6 +258,7 @@ func InitConfig(config pkgconfigmodel.Config) {
 	config.BindEnvAndSetDefault("health_port", int64(0))
 	config.BindEnvAndSetDefault("disable_py3_validation", false)
 	config.BindEnvAndSetDefault("python_version", DefaultPython)
+	config.SetKnown("win_skip_com_init")
```
Comment (pgimalac) [inline pkg/config/setup/config.go:261]: Let me know if I should set a default and/or bind environment variable

File: cmd/security-agent/subcommands/start/command.go (hunk 1 above)
Comment (paulcacheux) [reply cmd/security-agent/subcommands/start/command.go:388]: Definitely not on purpose, thanks !

**Review by paulcacheux — APPROVED** [review]

File: pkg/config/setup/config.go (hunk 3 above)
Comment (pgimalac) [reply pkg/config/setup/config.go:2046]: Seen with metrics-and-logs, `SetKnown` is fine

File: pkg/config/setup/config.go (hunk 5 above)
Comment (dustmop) [reply pkg/config/setup/config.go:369]: Setting a value of `0` works okay in this case. The secretResolver will see the zero and set its own default at configuration time: https://github.com/DataDog/datadog-agent/blob/e0297cc06d2b6d5b40fbaeed97dd6b3669407a6d/comp/core/secrets/secretsimpl/secrets.go#L188.

Comment (pgimalac) [reply pkg/config/setup/config.go]: Done !

File: pkg/config/setup/config.go (hunk 6 above)
Comment (pgimalac) [reply pkg/config/setup/config.go]: Discussed with Windows Agent, was told to use `BindEnvAndSetDefault`

**Review by brycekahle — APPROVED** [review]

**Review by dustmop — APPROVED** [review]

**Pull request merged**